ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -1925,6 +1926,11 @@ public RelNode apply(RelOptCluster cluster, RelOptSchema relOptSchema, SchemaPlu
         perfLogger.PerfLogEnd(this.getClass().getName(), PerfLogger.OPTIMIZER, "Calcite: Window fixing rule");
       }
 
+      perfLogger.PerfLogBegin(this.getClass().getName(), PerfLogger.OPTIMIZER);
```

Comment:
I believe we may still compute aggregations that are not used in the plan output. Could include the following test to confirm:
```
explain cbo
SELECT COUNT(*) FROM t1 ORDER BY SUM(col), COUNT(*);
```
I believe this is not critical, but we can add the test and a note to the q file, and create a placeholder JIRA.

Reply:
@jcamachor I am not sure if I understand the point fully.  In your example ORDER BY is already being removed. Are you suggesting to open JIRA to remove extra aggregate?

Reply:
Yes, that is what I meant.

Reply:
I believe now that the rule relies on the logic from the metadata provider that is more powerful, we could move this before step 6. in ```applyPreJoinOrderingTransforms```. This means that ```RelFieldTrimmer``` will be executed after this rule and thus may eliminate any column(s) that were used by SortLimit but are not needed anymore (for instance, the example above).

Reply:
@jcamachor I have moved the rule to `applyPreJoinOrderingTransforms`